prov/efa/src/efa_mr.c

```diff
@@ -305,6 +305,15 @@ static int efa_mr_dereg_impl(struct efa_mr *efa_mr)
 			ret = err;
 		}
 	}
+
+	if (efa_mr->peer.iface == FI_HMEM_CUDA && cuda_is_gdrcopy_enabled()) {
+		err = cuda_gdrcopy_dev_unregister(efa_mr->peer.gdrcopy_handle);
```

Comment:
Do we need to check if `gdrcopy_handle` is set? There are some error paths where this will get called without the register call completing, see L413 in this file.